packages/components/src/dropdown/test/index.js

```diff
@@ -1,98 +1,71 @@
 /**
  * External dependencies
  */
-import { fireEvent, render } from '@testing-library/react';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 
 /**
  * Internal dependencies
  */
-import Dropdown from '../';
-
-function getButtonElement( container ) {
-	return container.querySelector( 'button' );
-}
-function getOpenCloseButton( container, selector ) {
-	return container.querySelector( selector );
-}
+import Dropdown from '..';
 
 describe( 'Dropdown', () => {
-	function expectPopoverVisible( container, value ) {
-		const popover = container.querySelector( '.components-popover' );
-		if ( value ) {
-			expect( popover ).toBeTruthy();
-		} else {
-			expect( popover ).toBeFalsy();
-		}
-	}
-
-	it( 'should toggle the dropdown properly', () => {
-		const expectButtonExpanded = ( container, expanded ) => {
-			expect( container.querySelectorAll( 'button' ) ).toHaveLength( 1 );
-			expect( getButtonElement( container ) ).toHaveAttribute(
-				'aria-expanded',
-				expanded.toString()
-			);
-		};
-
-		const {
-			container: { firstChild: dropdownContainer },
-		} = render(
+	it( 'should toggle the dropdown properly', async () => {
+		render(
 			<Dropdown
-				className="container"
-				contentClassName="content"
 				renderToggle={ ( { isOpen, onToggle } ) => (
 					<button aria-expanded={ isOpen } onClick={ onToggle }>
-						Toggleee
+						Toggle
 					</button>
 				) }
-				renderContent={ () => <span>test</span> }
+				renderContent={ () => 'Content' }
 			/>
 		);
 
-		expectButtonExpanded( dropdownContainer, false );
-		expectPopoverVisible( dropdownContainer, false );
+		const user = userEvent.setup( {
+			advanceTimers: jest.advanceTimersByTime,
+		} );
+		const button = screen.getByRole( 'button', { expanded: false } );
 
-		const button = getButtonElement( dropdownContainer );
-		fireEvent.click( button );
+		expect( screen.queryByText( 'Content' ) ).not.toBeInTheDocument();
+		expect( button ).toBeVisible();
 
-		expectButtonExpanded( dropdownContainer, true );
-		expectPopoverVisible( dropdownContainer, true );
+		await user.click( button );
+		expect( button ).toHaveAttribute( 'aria-expanded', 'true' );
```

Comment:
We could alternatively make a new query for the button with `expanded: true` if we don't want to get into the internals like this.

Reply:
Ah. Yes, my mistake. I'll update.

Reply:
hmm. Does it make sense to create a new query when not using it in an `expect`?

or something like this:
```js
expect(
	screen.getByRole( 'button', { expanded: true } )
).toBeVisible();
```

Reply:
Yup something like that was what I had in mind 👍

Reply:
Thanks for the clarification! Updated